crates/astria-sequencer/src/app.rs

```diff
@@ -344,13 +351,15 @@ impl App {
     async fn execute_block_data(
         &mut self,
         txs: Vec<bytes::Bytes>,
+        mut remaining_cometbft_bytes: Option<usize>,
```

Comment:
I am not a fan of arguments of this type which fundamentally change the behavior of the contained code. When reading the code it's not obvious what the `None` argument is supposed to do.

IMO it's cleaner to have 2 separate functions `execute_block_data_with_limit` and `execute_block_data_without_limit` (bad names; feel free to bikeshed) that might internally delegate to `Some(capacity)` and `None` so that the purpose is clearer to the reader

Reply:
Cool, yeah it's def not great in this state. I'll refactor into freestanding functions